submission/distributor_test.go

```diff
@@ -308,33 +312,113 @@ func TestDistributorAddPreChain(t *testing.T) {
 		// TODO(merkulova): Add tests to cover more cases where log roots aren't available
 	}
 
+	for _, tc := range testCases {
+		for _, isPreCall := range []bool{true, false} {
+			pre := ""
+			if isPreCall {
+				pre = "Pre"
+			}
+			t.Run(pre+tc.name, func(t *testing.T) {
+				dist, _ := NewDistributor(tc.ll, tc.plc, newLocalStubLogClient, monitoring.InertMetricFactory{})
+				ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+				defer cancel()
+
+				if tc.getRoots {
+					if errs := dist.RefreshRoots(ctx); len(errs) != 1 || errs["ct.googleapis.com/icarus/"] == nil {
+						// 1 error is expected, because the Icarus log has an invalid root (see RootCerts).
+						t.Fatalf("dist.RefreshRoots() = %v, want 1 error for 'ct.googleapis.com/icarus/'", errs)
+					}
+				}
+
+				var scts []*AssignedSCT
+				var err error
+				pemChainFile := tc.pemChainFile
+				if isPreCall {
+					pemChainFile = strings.Replace(tc.pemChainFile, "subleaf", "subleaf-pre", 1)
```

Comment:
I think it'd be simpler to just have a `isPrecert` field in the testcase and duplicate all of them. That way, if a test fails, there'll always be a matching test case and it'll be obvious at a glance what its parameters were. Currently, you have to read through the test code to see that a different PEM file is being used to what the testcase definition claims.

Reply:
Done.